### PR TITLE
[15.0][MIG] privacy_consent: Migration to 15.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 15.0

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT39027

### CODE REVIEW

**Comment by victoralmau** [comment]
Ping @pedrobaeza and @chienandalu

File: privacy_consent/data/mail.xml [hunk 1]
```diff
@@ -23,8 +23,8 @@
                             <td>
                                 <a href="/">
                                     <img
-                                        src="/logo"
-                                        alt="${object.activity_id.controller_id.display_name|safe}"
+                                        t-attf-src="'/logo'"
```
Comment (pedrobaeza) [inline privacy_consent/data/mail.xml]: This one doesn't need `t-attf-src`

File: privacy_consent/data/mail.xml [hunk 2]
```diff
@@ -42,28 +42,33 @@
                                 style="padding:15px 20px 0px 20px; font-size:16px;"
                             >
                                 <p>
-                                    Hello, ${object.partner_id.name|safe}
+                                    Hello, <t t-out="object.partner_id.name" />
```
Comment (pedrobaeza) [inline privacy_consent/data/mail.xml:45]: Why not `Hello, {{object.partner_id.name}}`?

Comment (pedrobaeza) [reply privacy_consent/data/mail.xml:45]: And the same applies to the rest.

**Review by pedrobaeza — COMMENTED** [review]
Please compare it to #71 to see if there's something more to do.

File: privacy_consent/data/mail.xml (hunk 2 above)
Comment (victoralmau) [reply privacy_consent/data/mail.xml:45]: According to v15 guidelines and all odoo examples they cannot be used in mail body.

**Comment by victoralmau** [comment]
Changes done according to v16 changes.
Pre-commit error will be solved with https://github.com/OCA/data-protection/pull/73

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**